File: api/skeleton/handlers/warnings.js
const { randomUUID } = require('crypto');
const db = require('../db');

async function acceptWarning(req, res) {
  const { fortune_type_key: fortuneTypeKey, ip, user_agent: userAgent } = req.body || {};
  if (!fortuneTypeKey) {
    return res.status(400).json({ code: 'BAD_REQUEST', message: 'Missing fortune_type_key' });
  }

  const ft = await db.query(
    'SELECT id FROM fortune_types WHERE key = $1',
    [fortuneTypeKey]
  );
  if (ft.rows.length === 0) {
    return res.status(404).json({ code: 'NOT_FOUND', message: 'Fortune type not found' });
  }

  await db.query(
    'INSERT INTO warnings_acceptance (id, user_id, fortune_type_id, ip, user_agent) VALUES ($1, $2, $3, $4, $5)',
    [randomUUID(), req.userId, ft.rows[0].id, ip || null, userAgent || null]
  );

  return res.json({ ok: true });
}

module.exports = {
  acceptWarning,
};

File: backend/routes/warnings.py
from fastapi import APIRouter, Depends, HTTPException
from pydantic import BaseModel
from uuid import uuid4

from ..db import get_conn
from .security import get_user_id

router = APIRouter(prefix='/warnings', tags=['warnings'])


class WarningRequest(BaseModel):
    fortune_type_key: str
    ip: str | None = None
    user_agent: str | None = None


@router.post('/accept')
def accept_warning(payload: WarningRequest, user_id: str = Depends(get_user_id)):
    with get_conn() as conn:
        with conn.cursor() as cur:
            cur.execute('SELECT id FROM fortune_types WHERE key = %s', (payload.fortune_type_key,))
            row = cur.fetchone()
            if not row:
                raise HTTPException(status_code=404, detail='Fortune type not found')
            fortune_type_id = row[0]
            cur.execute(
                'INSERT INTO warnings_acceptance (id, user_id, fortune_type_id, ip, user_agent) VALUES (%s, %s, %s, %s, %s)',
                (str(uuid4()), user_id, fortune_type_id, payload.ip, payload.user_agent),
            )
            conn.commit()

    return {'ok': True}
